pkg/tsdb/loki/loki.go

```diff
@@ -122,16 +123,23 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 func getAuthHeadersForCallResource(headers map[string][]string) map[string]string {
 	data := make(map[string]string)
 
-	if auth := arrayHeaderFirstValue(headers["Authorization"]); auth != "" {
-		data["Authorization"] = auth
-	}
-
-	if cookie := arrayHeaderFirstValue(headers["Cookie"]); cookie != "" {
-		data["Cookie"] = cookie
-	}
+	for k, values := range headers {
+		k = textproto.CanonicalMIMEHeaderKey(k)
+		firstValue := arrayHeaderFirstValue(values)
 
-	if idToken := arrayHeaderFirstValue(headers["X-ID-Token"]); idToken != "" {
-		data["X-ID-Token"] = idToken
+		if firstValue == "" {
+			continue
+		}
+		switch k {
```

Comment:
i don't think you need the `continue` statements here. there is no more code after the `switch` block, and in go only the matches `case` is executed as far as i know ( https://go.dev/tour/flowcontrol/9 )
or maybe i'm missing something 😄

Reply:
Yes you are right. I didn't put too much thought after replicating what I saw on this PR: https://github.com/grafana/grafana/pull/58506
Should be fixed now.